pkg/plugins/runtime/k8s/controllers/gatewayapi/condition.go

```diff
@@ -15,18 +16,18 @@ func conditionOn(
 	}
 }
 
-func getCondition(deployment *kube_apps.Deployment, typ kube_apps.DeploymentConditionType) *metav1.ConditionStatus {
-	for _, c := range deployment.Status.Conditions {
-		if c.Type == typ {
-			status := metav1.ConditionStatus(c.Status)
+func getReady(instance *mesh_k8s.GatewayInstance) *metav1.ConditionStatus {
+	for _, c := range instance.Status.Conditions {
+		if c.Type == mesh_k8s.GatewayInstanceReady {
+			status := c.Status
 			return &status
 		}
 	}
 
 	return nil
 }
 
-func setConditions(gateway *gatewayapi.Gateway, deployment *kube_apps.Deployment) {
+func setConditions(gateway *gatewayapi.Gateway, instance *mesh_k8s.GatewayInstance) {
 	conditions := []metav1.Condition{
 		conditionOn(gateway, gatewayapi.GatewayConditionScheduled, metav1.ConditionTrue, gatewayapi.GatewayReasonScheduled),
 	}
```

Comment:
The conditions are a map keyed on the type, so I wonder whether this code would be easier to read if you used a `map[string]metav1Condition` and then flattened it to a slice later :shrug:

Reply:
It's rather frustrating that the serialization machinery isn't capable of handling maps and it has to be written manually every time.

I've cleaned things up a bit and really only `getReadyCondition` has to deal with the fact that it's a list.